kotlin-service/src/main/kotlin/com/microservice/kotlin/controller/TaskController.kt

```diff
@@ -3,49 +3,75 @@ package com.microservice.kotlin.controller
 import com.microservice.kotlin.model.Task
 import com.microservice.kotlin.repository.TaskRepository
 import io.swagger.annotations.ApiParam
+import org.apache.commons.lang.StringUtils
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.http.HttpStatus
 import org.springframework.http.MediaType
 import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PostFilter
 import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.core.authority.SimpleGrantedAuthority
 import org.springframework.web.bind.annotation.*
 import org.springframework.web.server.ResponseStatusException
 import java.net.URI
+import java.util.*
+import javax.validation.Valid
 
 @RestController
 @RequestMapping("/api/tasks")
-class TaskController {
-
-    @Autowired
-    lateinit var repository: TaskRepository
+class TaskController(@Autowired val repository: TaskRepository) {
 
     @GetMapping("/{id}")
     @PreAuthorize("hasAnyRole('ADMIN', 'TASK_READ', 'TASK_SAVE')")
-    fun findById(@PathVariable id: String): Task? = repository.findById(id)
+    fun findById(@PathVariable id: String,
+                 @AuthenticationPrincipal authentication : Authentication): Task = repository.findById(id)
+        .map {
+            if (authentication.authorities.contains(SimpleGrantedAuthority("ROLE_ADMIN")) || it.createdByUser == authentication.name) {
```

Comment:
also in kotlin you can declare [extension functions](https://kotlinlang.org/docs/tutorials/kotlin-for-py/extension-functionsproperties.html), for example add this to TaskController:

```
private fun Authentication.hasAdminAuthority() = SimpleGrantedAuthority("ROLE_ADMIN") in authentication.authorities
private fun Task.wasCreatedBy(name: String) = this.createdByUser == name
```

then this check can be replaced by the code below which is easier to read:

```
if (authentication.hasAdminAuthority() || task.wasCreatedBy(authentication.name)) {
```

assuming that you gave the name `task` to the lambda variable, otherwise use `it.wasCreatedBy`